### PR TITLE
CTC API for JAX

### PR DESCRIPTION
**Pull request by MaanasArora**
CTC loss implementation for JAX. I also included a documentation fix for the op.

Thank you!

### CODE REVIEW

File: keras/ops/nn.py [hunk 1]
```diff
@@ -1823,7 +1823,7 @@ def ctc_loss(target, output, target_length, output_length, mask_index=0):
     Args:
         target: A tensor of shape `(batch_size, target_max_length)` containing
             the true labels in integer format.
-        output: A tensor of shape `(batch_size, output_max_length, num_classes)`
+        output: A tensor of shape `(output_max_length, batch_size, num_classes)`
```
Comment (fchollet) [inline keras/ops/nn.py:1826]: That seems very counterintuitive given that all output tensors in Keras start with the batch dimension. Surely we should transpose?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/ops/nn.py (hunk 1 above)
Comment (MaanasArora) [reply keras/ops/nn.py]: Definitely, this is fixed now. Thanks!

File: keras/backend/jax/nn.py [hunk 2]
```diff
@@ -558,3 +558,78 @@ def batch_normalization(
...
+    target,
+    output,
+    target_length,
+    output_length,
+    mask_index=0,
+):
+    batch_size, _, _ = output.shape
+    batch_size, max_target_length = target.shape
+
+    output = output.transpose((1, 0, 2))
```
Comment (fchollet) [inline keras/backend/jax/nn.py:573]: Is this necessary for the scan op on the first dimension?

File: keras/backend/tensorflow/nn.py [hunk 3]
```diff
@@ -819,6 +819,7 @@ def ctc_loss(
     target = tf.cast(target, dtype="int32")
     output = tf.convert_to_tensor(output)
     output = tf.cast(output, dtype="float32")
+    output = tf.transpose(output, perm=(1, 0, 2))
```
[the remark this answers is not in the document]
Comment (MaanasArora) [reply keras/backend/tensorflow/nn.py]: Right, done, thanks!

File: keras/backend/jax/nn.py (hunk 2 above)
Comment (MaanasArora) [reply keras/backend/jax/nn.py:573]: Yes AFAIK, since we're scanning along the time dimension and scan runs along the leading axis.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
